drivers/at_cmd/at_cmd.c

```diff
@@ -48,6 +54,133 @@ struct callback_work_item {
 	at_cmd_handler_t callback;
 };
 
+/**@brief Find notification handler. */
+static notif_handler_t *find_notif_handler(at_cmd_handler_t handler)
+{
+	notif_handler_t *l_notif_handlers = notif_handlers;
```

Comment:
Before going further,

I'm not sure if you are aware, but Zephyr provides a single-linked list implementation:
https://github.com/zephyrproject-rtos/zephyr/blob/master/include/sys/slist.h

It provides quite a few convenience functions and is already used in a few places in NCS - any chance to use it here as well?

Just note that the API is not thread-safe, so synchronization primitives would be needed anyway.

Reply:
Used slist primitives, though it takes longer in the development process to double-check the behaviors of such implementation.

As mentioned [here](https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1163#discussion_r326614614), Semaphore has been used to protect slist.